_source/_posts/2021-12-16-cloud-native-java-microservices-with-istio.md

```diff
@@ -0,0 +1,580 @@
+
+### Create an Okta application
+
+{% include setup/cli.md type="jhipster" %}
+
+### Configure OpenID Connect settings to use Okta
+
+Open `src/main/resources/config/application.yml` file for each application and update `issuer-uri`, `client-id` and `client-secret` under the `spring.security.oauth2.client` section as shown below with values you obtained during previous step.
+
+```yml
+spring:
+  ...
+  security:
+    oauth2:
+      client:
+        provider:
+          oidc:
+            issuer-uri: https://{yourOktaDomain}/oauth2/default
+        registration:
+          oidc:
+            client-id: {yourClientId}
+            client-secret:
+            scope: openid,profile,email
+```
+
+> **Note**: For security reasons we will not set the client-secret on this file and instead use environment variable, `SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_OIDC_CLIENT_SECRET` or Kubernetes secrets. You can also skip updating the application.yml files altogether and use environment variable or Kubernetes secrets for all the OIDC configurations.
+
+You can run `./gradlew` on the store application to test the Okta login. Make sure to set the client secret environment variable first.
+
+## Create a GKE Cluster and Install Istio
+
+In order to deploy the stack to Google Kubernetes Engine, we need to create a cluster and install Istio. So lets begin by creating a cluster using Google CLoud SDK.
+
+### Create a cluster
+
+Make sure you are logged into the gcloud CLI and run the below command to create a GKE cluster.
+
+```bash
+# set region and zone
+$ gcloud config set compute/region europe-west1
+$ gcloud config set compute/zone europe-west1-b
+# Create a project and enable container APIs
+$ gcloud projects create jhipster-demo # You need to also enable billing via GUI
+$ gcloud config set project jhipster-demo
+$ gcloud services enable container.googleapis.com
+
+# Create GKE Cluster
+$ gcloud container clusters create hello-hipster \
+     --num-nodes 4 \
+     --machine-type n1-standard-2
+```
+
+This could take anywhere between 5-15 minutes. `--machine-type` is important as we need more CPU than what is available in the default setup. Once the cluster is created it should be set automatically as the current Kubernetes context. You can verify that by running `kubectl config current-context`. IF the new cluster is not set as the current context, you can set it by running `gcloud container clusters get-credentials hello-hipster`.
+
+{% img blog/cloud-native-java-microservices-with-istio/kdash-clusters.png alt:"GKE Cluster nodes" width:"900" %}{: .center-image }
+
+> **Note**: I'm using [KDash](https://kdash.cli.rs/) to monitor the cluster, you can try it or use kubectl, [k9s](https://github.com/derailed/k9s) and so on as you prefer.
```

Comment:
I would remove the > and just use **NOTE:**. If you want to have cool admonitions automatically, switch from markdown to Asciidoctor. ;)